node/node.go

```diff
@@ -743,6 +752,16 @@ func (node *AlgorandFullNode) GetPendingTxnsFromPool() ([]transactions.SignedTxn
 	return bookkeeping.SignedTxnGroupsFlatten(node.transactionPool.PendingTxGroups()), nil
 }
 
+// ensureParticipationDB opens or creates a participation DB.
+func ensureParticipationDB(genesisDir string, log logging.Logger) (account.ParticipationRegistry, error) {
+	accessorFile := filepath.Join(genesisDir, config.ParticipationRegistryFilename)
+	accessor, err := db.OpenPair(accessorFile, false)
+	if err != nil {
+		return nil, err
+	}
+	return account.MakeParticipationRegistry(accessor, log)
+}
+
```

Comment:
Ideally, this logic would also be part of the key manager/participation registry. But it doesn't have to be part of this PR.

Reply:
Maybe... maybe not. Passing in the accessor is nice since our db framework allows for in-memory databases, so testing is very easy. I could see hiding the participation registry in the account manager though.

Reply:
I wasn't suggesting that the key manager/participation registry would always be working with the read database accessor. I was just saying that the loading portion would all be in one place. It's perfectly fine to have key manager to allow the caller to modify/select a different accessor.